oneview/data_source_ethernet_network.go

```diff
@@ -0,0 +1,143 @@
+// (C) Copyright 2019 Hewlett Packard Enterprise Development LP
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package oneview
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceEthernetNetwork() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceEthernetNetworkRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"vlan_id": {
+				Type: schema.TypeInt,
+				//Required: true,
+				Computed: true,
+				ForceNew: true,
```

Comment:
I think ForcedNew is not required for data source, check this once.

Reply:
ForceNew: true is required as any change in the field will delete the resource and create new resource .

Reply:
Got it

Reply:
Since this is a datasource, we aren't deleting or creating anything. A datasource is just reading a resource. The only attribute that needs to specified are the attributes needed to obtain the resource, which in this case is the name. All other `Optional` tags for the other attributes can be removed.

Reply:
Updated as per comment